src/ILCompiler.CppCodeGen/src/CppCodeGen/ILToCppImporter.cs

```diff
@@ -1461,6 +2018,40 @@ private void ImportCalli(int token)
                 AppendLine();
                 Append(GetStackValueKindCPPTypeName(retKind, retType));
                 Append(" ");
+                Append(temp);
+                AppendSemicolon();
+            }
+            else
+            {
+                AppendLine();
+            }
+
+            var fnPtrValue = _stack.Pop();
+
+            string fatPtr = NewTempName();
+
+            Append("intptr_t ");
+            Append(fatPtr);
+            Append(" = ");
+            Append(fnPtrValue);
+
+            AppendSemicolon();
+
+            Append("if (");
+            Append(fatPtr);
+            Append(" & 0x2) {");
```

Comment:
Can you change this and below to use `FatFunctionPointerConstants.Offset` instead? You might need to add RuntimeConstants.cs to the project, but that should be fine.

(Also below.)